src/main/java/org/myrobotlab/service/Tracking.java

```diff
@@ -1,592 +1,395 @@
-          System.currentTimeMillis() - lastTimestamp, waitInterval));
-      lastTimestamp = System.currentTimeMillis();
+    if (config.pid != null) {
+      attachPid(config.pid);
     }
-
-    if (waitInterval < System.currentTimeMillis() - lastTimestamp) {
-      // setLocation(data);
-      // number of objects have stated the same
-      if (STATE_LEARNING_BACKGROUND.equals(state)) {
-        if (numberOfNewObjects == 0) {
-          // process background
-          // data.putAttribute(BACKGROUND);
-          data.put(PART, BACKGROUND);
-          invoke("toProcess", data);
-          // ready to search foreground
-          searchForeground();
-        }
-      } else {
-
-        if (numberOfNewObjects > 0) {
-          data.put(PART, FOREGROUND);
-          invoke("toProcess", data);
-        }
-      }
+    if (config.cv != null) {
+      attachCv(config.cv);
     }
 
-    lastNumberOfObjects = numberOfNewObjects;
-
-  }
-
-  public void connect(OpenCV opencv, ServoControl x, ServoControl y) {
-    log.info("Connect 2 servos for head tracking!... aye aye captain.  Also.. an open cv instance.");
-    attach(opencv);
-    attach(x, y);
-    // don't understand this, it should be getMin and getMax, no ? but
-    // seem worky..
-    pid.setOutputRange("x", -x.getMax(), x.getMax());
-    pid.setOutputRange("y", -y.getMax(), y.getMax());
-    // target the center !
-    pid.setSetpoint("x", 0.5);
-    pid.setSetpoint("y", 0.5);
-    // TODO: remove this sleep stmt
-    sleep(100);
-    rest();
-  }
-
-  public void attach(ServoControl servo, String axis) {
-    if (!(axis.equals("x") || axis.equals("y"))) {
-      log.info("Axis must be x or y");
-      return;
+    if (config.enabled) {
+      // enable();
+    } else {
+      disable();
     }
-    TrackingServoEvent tsd = new TrackingServoEvent(axis);
-    tsd.servoControl = servo;
-    tsd.axis = axis;
-    servoControls.put(axis, tsd);
-    tsd.currentServoPos = servo.getCurrentInputPos();
-    log.info("Tracking attach : Axis : {} Servo {} current position {}", axis, servo.getName(), servo.getCurrentInputPos());
+    return config;
   }
-
-  public void attach(ServoControl x, ServoControl y) {
-    attach(x, "x");
-    attach(y, "y");
+  
+  public boolean isIdle() {
+    return state == TrackingState.IDLE;
   }
 
-  public void attach(OpenCV opencv) {
-    this.opencv = opencv;
-    opencv.addListener("publishOpenCVData", getName(), "onOpenCVData");
-  }
+  public void startWithDefaults() {
+    if (pan == null) {
+      pan = String.format("%s.pan", getName());
+      Runtime.start(pan, "Servo");
+      attachPan(pan);
+    }
+    if (tilt == null) {
+      tilt = String.format("%s.tilt", getName());
+      Runtime.start(tilt, "Servo");
+      attachTilt(tilt);
+    }
+    if (pid == null) {
+      pid = String.format("%s.pid", getName());
+      Runtime.start(pid, "Pid");
+      attach(pid);

```

Comment:
this should configure default pid settings here.  the servo names are known in this scenario.